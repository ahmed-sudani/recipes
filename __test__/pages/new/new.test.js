import { render } from '@testing-library/react'
import Create from '../../../pages/new'

describe('New page', () => {
  it('should render correctly', () => {
    const { container } = render(<Create />)
    expect(container).toMatchSnapshot()
  })
})
