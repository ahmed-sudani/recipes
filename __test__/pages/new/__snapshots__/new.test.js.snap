// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New page should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="image"
    >
      <svg
        fill="currentColor"
        height="200"
        viewBox="0 0 16 16"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 11a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
        />
        <path
          d="M13.997 5.17a5 5 0 0 0-8.101-4.09A5 5 0 0 0 1.28 9.342a5 5 0 0 0 8.336 5.109 3.5 3.5 0 0 0 5.201-4.065 3.001 3.001 0 0 0-.822-5.216zm-1-.034a1 1 0 0 0 .668.977 2.001 2.001 0 0 1 .547 3.478 1 1 0 0 0-.341 1.113 2.5 2.5 0 0 1-3.715 2.905 1 1 0 0 0-1.262.152 4 4 0 0 1-6.67-4.087 1 1 0 0 0-.2-1 4 4 0 0 1 3.693-6.61 1 1 0 0 0 .8-.2 4 4 0 0 1 6.48 3.273z"
        />
      </svg>
    </div>
    <form
      action="/api/recipes"
      class="column"
      method="POST"
    >
      <div
        class="Text"
      >
        Add new recipe to the site
      </div>
      <div
        class="inputForm"
      >
        <label>
           
          Name
           
        </label>
        <input
          class="input"
          maxlength="50"
          minlength="5"
          name="Name"
          required=""
          type="text"
        />
      </div>
      <div
        class="inputForm"
      >
        <label>
           
          Image Url
           
        </label>
        <input
          class="input"
          maxlength="100"
          name="Image Url"
          required=""
          type="url"
        />
      </div>
      <div
        class="inputForm"
      >
        <label>
           
          Country
           
        </label>
        <input
          class="input"
          maxlength="50"
          minlength="2"
          name="Country"
          required=""
          type="text"
        />
      </div>
      <div
        class="inputForm"
      >
        <label>
           
          Required Time
           
        </label>
        <input
          class="input"
          max="180"
          min="1"
          name="Required Time"
          required=""
          type="number"
        />
      </div>
      <div
        class="inputForm"
      >
        <label>
          Ingredients
        </label>
        <div
          class="ingredientsInput"
        >
          <input
            type="text"
          />
          <button
            type="button"
          >
            <svg
              fill="currentColor"
              height="30"
              viewBox="0 0 16 16"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
              />
              <path
                d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
              />
            </svg>
          </button>
        </div>
      </div>
      <button
        class="btn"
      >
        Create Recipe
      </button>
    </form>
    <ul
      class="list"
    />
  </div>
</div>
`;
